lib/mini_profiler/snapshots_transporter.rb

```diff
@@ -50,12 +51,24 @@ def flush_buffer
       @buffer.dup if @buffer.size > 0
     end
     if buffer_content
-      request = Net::HTTP::Post.new(
-        @uri,
+      headers = {
         'Content-Type' => 'application/json',
         'Mini-Profiler-Transport-Auth' => @auth_key
-      )
-      request.body = { snapshots: buffer_content }.to_json
+      }
+      json = { snapshots: buffer_content }.to_json
+      body = if @gzip_requests
+        require 'zlib'
+        io = StringIO.new
+        gzip_writer = Zlib::GzipWriter.new(io)
+        gzip_writer.write(json)
+        gzip_writer.close
+        headers['Content-Encoding'] = 'gzip'
```

Comment:
oh I missed this line...  This change looks fine to me, I do wonder though if we need this toggle `snapshots_transport_gzip_requests` ... I guess it teaches people that you can compress.

Approving!